src/main/java/yams/config/UnitsParser.java

```diff
@@ -0,0 +1,667 @@
+import static edu.wpi.first.units.Units.Volts;
+
+import edu.wpi.first.units.measure.Angle;
+import edu.wpi.first.units.measure.AngularAcceleration;
+import edu.wpi.first.units.measure.AngularVelocity;
+import edu.wpi.first.units.measure.Current;
+import edu.wpi.first.units.measure.Distance;
+import edu.wpi.first.units.measure.LinearAcceleration;
+import edu.wpi.first.units.measure.LinearVelocity;
+import edu.wpi.first.units.measure.Mass;
+import edu.wpi.first.units.measure.Time;
+import edu.wpi.first.units.measure.Voltage;
+
+/**
+ * A class that converts a magnitude and a unit into a {@link Measurement}
+ * object.
+ */
+public class UnitsParser {
+    public final static String M = "m";
+    public final static String CM = "cm";
+    public final static String MM = "mm";
+    public final static String IN = "in";
+    public final static String FT = "ft";
+    public final static String YD = "yd";
+    public final static String MPS = "m/s";
+    public final static String CMPS = "cm/s";
+    public final static String MPS2 = "m/s^2";
+    public final static String CMPS2 = "cm/s^2";
+    public final static String DEG = "deg";
+    public final static String RAD = "rad";
+    public final static String DEGPS = "deg/s";
+    public final static String RADPS = "rad/s";
+    public final static String DEGPS2 = "deg/s^2";
+    public final static String RADPS2 = "rad/s^2";
+    public final static String AMPS = "amps";
+    public final static String VOLTS = "volts";
+    public final static String SEC = "sec";
+    public final static String MS = "ms";
+    public final static String US = "us";
+    public final static String NS = "ns";
+    public final static String KG = "kg";
+    public final static String G = "g";
+    public final static String MG = "mg";
+    public final static String OZ = "oz";
+    public final static String LBS = "lbs";
+    public final static String STONE = "stone";
+    public final static String TONS = "tons";
+    private static final String FPS = "ft/s";
+    private static final String FPS2 = "ft/s^2";
+
+    /**
+     * Converts a magnitude and a unit into a {@link Distance} object.
+     * 
+     * @param unitValueJson A json object containing the magnitude and unit of the
+     *                      distance.
+     * @return The {@link Distance} object.
+     */
+    public static Distance parseDistance(UnitValueJson unitValueJson) {
+        return parseDistance(unitValueJson.val, unitValueJson.uom);
+    }
+
+    /**
+     * Converts a magnitude and a unit into a {@link Distance} object.
+     * 
+     * @param magnitude The magnitude of the distance.
+     * @param unit      The unit of the distance. Can be "meters", "feet", "inches",
+     *                  "mm", "cm", or "yd".
+     *                  (case insensitive)
+     * @return The {@link Distance} object.
+     */
+    public static Distance parseDistance(double magnitude, String unit) {
+        switch (unit.trim().toLowerCase()) {
```

Comment:
While i appreciate this, i feel like it would be more efficient to make this a `Map<List<String>,Supplier>`?